packages/docs/connections/Elasticsearch.yaml

```diff
@@ -0,0 +1,156 @@
+            - `node: string|string[]`: __Required__ - Elasticsearch node(s) to connect to.
+            - `index: string`: The default index to query. You can also provide a per-request index to override this setting.
+            - `auth: BasicAuth|CloudAuth`: The authentication data as an object, containing either basic auth credentials (as `username` and `password`), an API key (as `apiKey`), or Elastic Cloud credentials.
+            - `maxRetries: number`: Max number of retries for each request.
+            - `requestTimeout: number`: Max request timeout in milliseconds for each request.
+
+            The connection accepts many more advanced configuration options. They will be passed to the Elasticsearch client verbatim, so check out the [configuration instructions](https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/basic-config.html) provided by Elastic.
+
+            #### Examples
+
+            ###### Authenticated connection:
+            ```yaml
+            connections:
+              - id: elasticsearch
+                type: Elasticsearch
+                properties:
+                  node: ELASTICSEARCH_HOST
+                  auth:
+                    username: ELASTICSEARCH_USER
+                    password: ELASTICSEARCH_PASSWORD
+            ```
+            Environment variables:
+            ```
+            LOWDEFY_SECRET_ELASTICSEARCH_HOST = http://localhost:9200
+            LOWDEFY_SECRET_ELASTICSEARCH_USER = es
+            LOWDEFY_SECRET_ELASTICSEARCH_PASSWORD = password
+            ```
+
+            ###### Unauthenticated connection with default index:
+            ```yaml
+            connections:
+              - id: elasticsearch
+                type: Elasticsearch
+                properties:
+                  node: ELASTICSEARCH_HOST
+                  index: ELASTICSEARCH_INDEX
+            ```
+            Environment variables:
+            ```
+            LOWDEFY_SECRET_ELASTICSEARCH_HOST = http://localhost:9200
+            LOWDEFY_SECRET_ELASTICSEARCH_INDEX = articles
+            ```
+
+            ## Requests
+
+            Request types:
+              - ElasticsearchSearch
+
+            ### ElasticsearchSearch
+
+            The `ElasticsearchSearch` request searches the Elasticsearch index.
+
+            #### Properties
+            - `query: object`: The Elasticsearch query, expressed in the [JSON Query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl.html). If you need to pass custom request body attributes, you should wrap the query within another `query` property (in that case, the top-level `query` object will be used as `body`).
+            - `aggs: object[]`: The [aggregations](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations.html) to request from the server.
+            - `_source: string|string[]|boolean`: True or false to return the _source field or not, or a list of fields to return.
```

Comment:
That's a tough one, didn't think about that yet. I guess it's easier to rename the request properties to `source`, `source_include`, and `source_exclude`, then map them into the Elasticsearch client parameters (inside the resolver all operators should have been resolved already, right?).  
It's pretty obvious for users, I hope.

On the other hand, Elasticsearch has several properties in the responses that are also prefixed with an underscore (eg. `_id`, `_score`, `_source`, etc.), don't know about that..

Reply:
Just remembered that this is less of an issue than I thought. So we have 2 rules with operators:
1) starts with _
2) the operator object must only have 1 key

Thus for implementations like these the second rule will mostly cause the operator not to evaluate for that object. So there is a much smaller chance of name clashes here. Not really sure how / where to document this.
For now at least I think we can keep this as is, right @SamTolmay ?

Reply:
I think it would be best to keep it as is, there should be at least another field like query specified, for the request to work. So this happening should be an edge case. We can also add a warning to the docs if we want.

Reply:
Yeah, `_source` on its own doesn't produce a valid query, so that'd probably be a bug anyway.